subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -92,7 +92,9 @@ static union {
 	struct hawkbit_cancel cancel;
 } hawkbit_results;
 
-static struct k_work_delayable hawkbit_work_handle;
+static void autohandler(struct k_work *work);
+
+static K_WORK_DELAYABLE_DEFINE(hawkbit_work_handle, autohandler);
```

Comment:
so that it isn't initialised in `hawkbit_autohandler()` and `hawkbit_autohandler()` can be executed more that one without reinitialising again. For maybe converting hawkBit to be instance-based, which will need lot of changes, initialising it statically is not the problem, it would be this being a global variable and that's not changing with this PR. Also making hawkbit instance-based would not be the optimal solution, as hawkBit already supports having multiple software modules as part of a distribution, where one module is the OS and another can be the for settings. That would also be much easier to implement.